tests/test_gmsh_IO.py

```diff
@@ -318,3 +318,89 @@ def test_generate_mesh(self):
 
         # check if the coordinates of the points are correct
         TestUtils.assert_dictionary_almost_equal(expected_mesh_data, mesh_data)
+
+    def test_geometry_data_2D(self):
+        """
+        Checks whether mesh data generated for 2D geometries is not empty.
+    """
+        # define the default mesh size
+        default_mesh_size = -1
+        # define the points of the surface as a list of tuples
+        input_points_list = [[(0, 0, 0), (3, 0, 0), (3, 1, 0), (0, 1, 0)],
+                             [(3, 0, 0), (5, 0, 0), (5, 1, 0), (4, 1.5, 0), (3, 1, 0)],
+                             [(0, 1, 0), (2, 1, 0), (2, 3, 0), (0, 3, 0)],
+                             [(2, 1, 0), (3, 1, 0), (4, 1.5, 0), (5, 1, 0), (5, 3, 0), (2, 3, 0)],
+                             [(0, 3, 0), (2.5, 3, 0), (2, 4, 0), (0, 4, 0)],
+                             [(0.8, 4, 0), (1.2, 4, 0), (1.2, 4.1, 0), (0.8, 4.1, 0)]]
+        # define the name labels for the surfaces
+        name_label_list = ["First Soil Layer", "FSL", "Second Soil Layer", "SSL", "Soil Ballast", "Line Track"]
+
+        # define geometry dimension; input "3" for 3D to extrude the 2D surface, input "2" for 2D
+        dims = 2
+        # if 3D, input depth of geometry to be extruded from 2D surface
+        extrusion_length = [0, 0, 0]
+        # if "True", saves mesh data to separate mdpa files; otherwise "False"
+        save_file = False
+        # if "True", opens gmsh interface; otherwise "False"
+        open_gmsh_gui = False
+        # set a name for mesh output file
+        mesh_output_name = "test_2D"
+        # set output directory
+        mesh_output_dir = "."
+
+        gmsh_io = GmshIO()
+
+        gmsh_io.generate_geometry(input_points_list, extrusion_length, dims,
+                                  mesh_output_name, name_label_list, default_mesh_size)
+        gmsh_io.generate_extract_mesh(dims, mesh_output_name, mesh_output_dir, save_file, open_gmsh_gui)
+
+        geo_data = gmsh_io.geo_data
+
+        assert geo_data is not None
+
+        for i in geo_data["physical_groups"].keys():
```

Comment:
can you check the actual content of physical_groups. As it is small, it should be doable